treeherder/model/models.py

```diff
@@ -1477,11 +1525,69 @@ class Meta:
         db_table = "text_log_error"
         unique_together = ('step', 'line_number')
 
+    def __str__(self):
+        return "{0} {1}".format(self.id, self.step.job.id)
+
+    @property
+    def metadata(self):
+        try:
+            return self._metadata
+        except TextLogErrorMetadata.DoesNotExist:
+            return None
+
     def bug_suggestions(self):
         from treeherder.model import error_summary
         return error_summary.bug_suggestions_line(self)
 
 
+class TextLogErrorMetadata(models.Model):
+    class Meta:
+        db_table = "text_log_error_metadata"
+
+    text_log_error = models.OneToOneField(TextLogError,
+                                          primary_key=True,
+                                          related_name="_metadata",
+                                          on_delete=models.CASCADE)
+
+    failure_line = models.OneToOneField(FailureLine,
+                                        related_name="text_log_error_metadata",
+                                        null=True)
+
+    # Note that the case of best_classification = None and best_is_verified = True
+    # has the special semantic that the line is ignored and should not be considered
+    # for future autoclassifications.
+    best_classification = models.ForeignKey("ClassifiedFailure",
+                                            related_name="best_for_errors",
+                                            null=True,
+                                            db_index=True,
```

Comment:
It is not necessary to set db_index on foreign keys, an index is automatically generated by default.